src/main/java/mate/controller/CreateNewDriverController.java

```diff
@@ -0,0 +1,33 @@
+package mate.controller;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Driver;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/drivers/create")
+public class CreateNewDriverController extends HttpServlet {
+    private static final Injector injector = Injector.getInstance("mate");
+    private final DriverService driverService
+            = (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/drivers/createNewDriver.jsp").forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException {
+        Driver driver = new Driver(req.getParameter("name"),
+                req.getParameter("licenseNumber"));
+        driverService.create(driver);
+        resp.sendRedirect("/drivers");
```

Comment:
And I'm not sure if it's a good to handle two types of requests in one servlet class because we had this point in the common mistakes:
**Remember that single controller should be responsible for a single task.**

Reply:
explain this point please. there's only one single task - creating a new driver. as you can see on videos it's a common situation to have doGet and doPost methods in one controller